### DISCUSSION TITLE
Using single digit days and ordinals for the daily note format?

### DISCUSSION THREAD

**Discussion by MoltenMonster** [epwalsh/obsidian.nvim#738] [Q&A]
In Obsidian under the Daily Note core plugin, I have the date format set to `YYYY-MM-DD MMMM Do YYYY dddd` which produces `2024-10-03 October 3rd 2024 Thursday`, and have lots of daily notes formatted like that. Looking here https://www.lua.org/pil/22.1.html, Iua doesn't seem to have a date format option for single digit dates, like `October 3` rather than `October 03`, or ordinals, like `3rd` rather than just 3.

any ideas for what to do with the `daily_notes` `date_format` option?

**Comment by azinsharaf** [comment]
This function generates a date string like 2024-October-3rd 2024 Thursday. you might need to tweak it a little bit. 

```
-- Function to add ordinal suffixes to a number
local function add_ordinal_suffix(day)
    if day % 100 >= 11 and day % 100 <= 13 then
        return day .. "th"
    else
        local suffix = { "st", "nd", "rd", "th" }
        return day .. suffix[(day % 10) > 0 and (day % 10) or 4]
    end
end

-- Custom date format function
local function custom_date_format()
    local year = os.date("%Y")
    local month_name = os.date("%B")
    local day = tonumber(os.date("%d"))
    local day_ordinal = add_ordinal_suffix(day)
    local weekday = os.date("%A")

    -- Combine the formatted parts
    return string.format("%s-%s-%s %s %s", year, month_name, day_ordinal, year, weekday)
end

-- Test the function
print(custom_date_format())
```
